### PR TITLE
Curve + yearn v2

### PR DESCRIPTION
**Pull request by yabirgb**
Add curve lp assets
Add yearnv2 vaults

### CODE REVIEW

**Comment by yabirgb** [comment]
The curve entries are autogenerated by reading the curve contracts and yearn from their graph for v2

File: updates/6/updates.sql [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+*
+INSERT INTO ethereum_tokens(address, decimals, protocol) VALUES("0xe9Dc63083c464d6EDcCFf23444fF3CFc6886f6FB", 18, "yearn_v2_vault"); INSERT INTO assets(identifier, type, name, symbol,started, swapped_for, coingecko, cryptocompare, details_reference) VALUES("_ceth_0xe9Dc63083c464d6EDcCFf23444fF3CFc6886f6FB", "C", "Curve oBTC Pool yVault", "yvCurve-oBTC", 1619129210, NULL, "", "", "0xe9Dc63083c464d6EDcCFf23444fF3CFc6886f6FB"); INSERT INTO underlying_tokens_list(address, weight, parent_token_entry) VALUES ("0x2fE94ea3d5d4a175184081439753DE15AeF9d614", "1.0", "0xe9Dc63083c464d6EDcCFf23444fF3CFc6886f6FB");
```
Comment (LefterisJP) [inline updates/6/updates.sql:221]: Damn these are way too many assets to review manually. How would you suggest I test them?

How did you autogenerate the correct start_timestamp? The others I guess can come from the chain.

File: validator/checker.py [hunk 2]
```diff
@@ -206,8 +207,10 @@ def check_single_version_update(
             if insert:
                 assert asset_data.cryptocompare not in seen_elements, f'Duplicate cryptocompare {asset_data.cryptocompare}'
                 assert asset_data.coingecko not in seen_elements, f'Duplicate coingecko {asset_data.coingecko}'
-                assert asset_data.name not in seen_elements, f'Duplicate name {asset_data.name}'
-                assert asset_data.symbol not in seen_elements, f'Duplicate symbol {asset_data.symbol}'
+                if asset_data.name in seen_elements:
```
Comment (LefterisJP) [inline validator/checker.py:210]: Why this change?

Comment (yabirgb) [reply validator/checker.py:210]: Because there are two vaults with the same name and symbol

Comment (LefterisJP) [reply validator/checker.py:210]: This is not a good change. You are just adding a log warning which does nothing in tests. So this turns the check to a noop. What you should do is add an exception for these specific vaults only

File: updates/6/updates.sql (hunk 1 above)
Comment (yabirgb) [reply updates/6/updates.sql:221]: The timestamp is not correct as I guess there is no way to know it. Does it matter since there is no queries for cc/coingecko?

Comment (yabirgb) [reply updates/6/updates.sql:221]: The data comes from calling the curve registry and information on chain so I believe it can be trusted. The SQL queries works fine since I've been using them in my branch

File: validator/checker.py (hunk 2 above)
Comment (yabirgb) [reply validator/checker.py:210]: okey

File: updates/6/updates.sql (hunk 1 above)
Comment (LefterisJP) [reply updates/6/updates.sql:221]: > The timestamp is not correct as I guess there is no way to know it. Does it matter since there is no queries for cc/coingecko?

The timestamp is used for log querying. It's important to be correct for ethereum contracts. Since it specifies when the log queries should start from.

Finding the timestamp is also probably possible automatically but it's a bit more complicated if I recall.

File: validator/checker.py [hunk 3]
```diff
@@ -207,10 +244,10 @@ def check_single_version_update(
             if insert:
                 assert asset_data.cryptocompare not in seen_elements, f'Duplicate cryptocompare {asset_data.cryptocompare}'
                 assert asset_data.coingecko not in seen_elements, f'Duplicate coingecko {asset_data.coingecko}'
-                if asset_data.name in seen_elements:
-                    logging.warning(f'Duplicate name {asset_data.name}')
-                if asset_data.symbol in seen_elements:
-                    logging.warning(f'Duplicate symbol {asset_data.symbol}')
+                if asset_data.name not in DUPLICATED_NAMES:
```
Comment (LefterisJP) [inline validator/checker.py:247]: Also add a comment here, why these symbol/names are duplicated. I am not even sure why. v1/v2 ?

Comment (yabirgb) [reply validator/checker.py:247]: In V2 anyone can create a vault. Then there are some that are endorsed by yearn once they are battle tested. The information I used comes from the yearn v2 graph that tracks every yearn v2 vault, even the ones that are not endorsed. 

Since this is all the information provided thought that it was a good idea to save all of them. One thing I can do is filter to only get the ones in this list https://tokenlists.org/token-list?url=https://yearn.science/static/tokenlist.json

Comment (LefterisJP) [reply validator/checker.py:247]: So perhaps add a similar comment here? I think if a user uses a v2 vault we should detect it, even if it's not endorsed.

Comment (yabirgb) [reply validator/checker.py:251]: Done

File: validator/checker.py [hunk 4]
```diff
@@ -50,6 +51,47 @@ class AssetData(NamedTuple):
     protocol: Optional[str]
     decimals: Optional[int]
 
+# In V2 anyone can create a vault and there are some that are endorsed by yearn once they 
+# are battle tested. The information I used comes from the yearn v2 graph that tracks every
```
Comment (LefterisJP) [inline validator/checker.py:55]: nitpick: Don't use first person in code.
```suggestion
# are battle tested. The information seen here comes from the yearn v2 graph that tracks every 
```

**Review by LefterisJP — COMMENTED** [review]
So this is too many tokens to actually check properly. The question on my side is will these cause duplicate balances? Since they will be part of the DB they will also be queried as token balances. Which means you will have to put exception in zerion for all of them to not cause duplicates.

**Comment by yabirgb** [comment]
@LefterisJP Wouldn't it better to ignore Yearn Vaults? At the moment we have https://github.com/rotki/rotki/blob/develop/rotkehlchen/chain/manager.py#L124

**Comment by LefterisJP** [comment]
> @LefterisJP Wouldn't it better to ignore Yearn Vaults? At the moment we have https://github.com/rotki/rotki/blob/develop/rotkehlchen/chain/manager.py#L124
> 
> Edit: Add an exception there along the other

That's what I mean by exception. But not sure if this covers allof them. You would have to look at all zerion protocosl to see if there is any more. Like a yearnv2 or something.

**Review by LefterisJP — APPROVED** [review]

File: validator/checker.py [hunk 5]
```diff
@@ -3,6 +3,7 @@
 https://github.com/rotki/rotki/blob/8f41fcba4732fba4af563c45c34ad9ad288906a2/rotkehlchen/globaldb/updates.py
 """
 
+import logging
```
Comment (LefterisJP) [inline validator/checker.py:6]: Hey @yabirgb I just noticed something. You are using the actual logging module. For pytest it's better to use 

`test_warnings` as we do for example here:  https://github.com/rotki/rotki/blob/f053a697d688035e3d3f6bce2d16733d02fa10ed/rotkehlchen/tests/unit/test_assets.py#L134-L139
